extensions/hashicorp-vault/src/test/java/org/eclipse/dataspaceconnector/core/security/hashicorpvault/HashicorpVaultIntegrationTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *  Copyright (c) 2022 Mercedes-Benz Tech Innovation GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Mercedes-Benz Tech Innovation GmbH - Initial Test
+ *
+ */
+
+package org.eclipse.dataspaceconnector.core.security.hashicorpvault;
+
+import org.eclipse.dataspaceconnector.common.util.junit.annotations.IntegrationTest;
+import org.eclipse.dataspaceconnector.junit.extensions.EdcExtension;
+import org.eclipse.dataspaceconnector.spi.security.Vault;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.Map;
+import java.util.UUID;
+
+import static org.eclipse.dataspaceconnector.core.security.hashicorpvault.HashicorpVaultExtension.VAULT_TOKEN;
+import static org.eclipse.dataspaceconnector.core.security.hashicorpvault.HashicorpVaultExtension.VAULT_URL;
+
+@IntegrationTest
+@Tag("HashicorpVaultIntegrationTest")
+@ExtendWith(EdcExtension.class)
+class HashicorpVaultIntegrationTest {
+    private static final String VAULT_TEST_URL = "http://127.0.0.1:8200";
+    private static final String VAULT_TEST_TOKEN = "test-token";
+
+    @BeforeEach
+    final void setup(EdcExtension extension) {
+        extension.setConfiguration(Map.of(VAULT_URL, VAULT_TEST_URL, VAULT_TOKEN, VAULT_TEST_TOKEN));
+    }
+
+    @Test
+    @DisplayName("Resolve a secret that exists")
+    void testResolveSecret_exists(Vault vault) {
+        var key = UUID.randomUUID().toString();
+        var valueExpected = UUID.randomUUID().toString();
+
+        vault.storeSecret(key, valueExpected);
+        var secretValue = vault.resolveSecret(key);
+        Assertions.assertEquals(valueExpected, secretValue);
```

Comment:
nit: static imports, and use AssertJ (`assertThat(xyz).isNull()`) instead for consistency